src/main/java/org/spongepowered/mod/mixin/core/entity/MixinEntity.java

```diff
@@ -111,6 +111,11 @@ public boolean teleportEntity(net.minecraft.entity.Entity entity, Location locat
             entityplayermp1.theItemInWorldManager.setWorld(toWorld);
             entityplayermp1.addSelfToInternalCraftingInventory();
             entityplayermp1.setHealth(entityplayermp1.getHealth());
+            
+            for(Object effect : entityplayermp1.getActivePotionEffects()) {
+                entityplayermp1.playerNetServerHandler.sendPacket(new S1DPacketEntityEffect(entityplayermp1.getEntityId(), (PotionEffect) effect));
```

Comment:
Editing on the web is probably not the smartest idea! Imports fixed, should all be fine now.